### ISSUE TITLE
Feature Request: Observe/Subscribe node

### ISSUE THREAD

**Issue by alexschwantes** [nidayand/node-red-contrib-tradfri#8]
Libcoap seems to offer the observe/subscribe capability, such that you could add a new node that subscribes to changes to either anything or a specific id and outputs the json.

**Comment by alexschwantes** [comment]
Also, what about using node-coap (https://github.com/mcollina/node-coap) instead of the external library. This would make installation easier as its all done within node...

**Comment by donster22** [comment]
Yes, this is something I’ve come searching for today. I’d like to know when the gateway does something so that I can have nodered do something else. Is that what the observe / subscribe capability would be used for as mentioned above?

**Comment by thunder-62** [comment]
I am currently polling the cateway for the bulb status but an observe/subscribe would be far more elegant and waste less ressources

**Comment by donster22** [comment]
Yes, for me it would allow almost real-time button status in the Dashboard UI and maybe I could repurpose motion sensors and switches?

**Comment by thunder-62** [comment]
all just FYI: polling is not a good Idea. I was polling the 14 bulbs that I have round robin with 5 seconds pause. this KILLED the gateway. I was able to revive the GW with a power cycle but once I had to factory reset and re-establish all connections.

One more thing: after the factory reset I was stuck with a 1.04 firmware release and the gateway did not update to the current firmware but found the lamps. at this stage I was not able to connect to the GW with node-red at all. I did another factory reset, the lamps were not connected anymore and then the GW was again reachable with node-red. (thought it micht help someone...)

So the observe/subscribe would be great...  Actually, I would try to contribute but I do not have any idea on the protocol elements needed for the observe... Any links?

**Comment by leifnel** [comment]
It would be nice to get the button pressed on the remote, perhaps even detaching the remote from the lights, so node-red processes the data and sends it to the lights.

Also, IKEA trådfri now have a motion detect; this would be good to get into node-red too.

**Comment by ruudboon** [comment]
Just a big thumbs up if this could be implemented. Thnx for this great piece of software

**Comment by spiu16** [comment]
having the same issue: I would like to use the remote for other purposes via nodered, therefor it would be great to receive the signals from the remote via the tradfri-out node.
@nidayand do you think this is possible?
Thanks!

**Comment by nidayand** [comment]
It will require a redesign of the complete enchiladas to use another library for talking to the devices and I'm sorry to say that it is currently low on my list of priorities right now..
